src/dcat_ap_no_validator_service/service/validator_service.py

```diff
@@ -13,19 +13,18 @@ class ValidatorService:
 
     def __init__(self, graph: str, version: str = "2") -> None:
         """Initialize service instance."""
-        self._g = Graph().parse(data=graph, format="turtle")
-        # TODO get version from user and map to correct shacl shape:
         logging.debug(f"Got request for version: {version}")
+        self._g = Graph().parse(data=graph, format="turtle")
         self._version = version
 
     async def validate(self) -> Tuple[bool, Graph, Graph, str]:
         """Validate function."""
         if len(self._g) == 0:  # No need to validate empty graph
-            raise Exception  # TODO: create proper Exception
+            raise ValueError
```

Comment:
Skulle vi ikke sagt litt mer om feilen? For eks ValueError('Graph cannot be empty').